File: source/livana-fe/src/types/response/homeListingResponse.ts
export interface FacilityQuantityDto {
  quantity: number;
  facilityId: string;
}

export interface ImageOrderResponse {
  image: {
    url: string;
    publicId: string;
  };
  order: number;
}

export interface ListingHostDto {
  hostId: string;
  hostDisplayName: string;
  avatarUrl: string;
  phoneNumber: string;
}

export interface HomeListingResponse {
  listingId: string;
  host: ListingHostDto;
  title: string;
  price: number;
  description: string;
  capacity: number;
  address: string;
  latitude: number;
  longitude: number;
  propertyTypeId: string;
  amenityIds: string[];
  facilities: FacilityQuantityDto[];
  images: ImageOrderResponse[];
}

File: source/livana-be/src/main/java/octguy/livanabe/dto/response/HomeListingResponse.java
package octguy.livanabe.dto.response;

import lombok.Builder;
import lombok.Data;
import octguy.livanabe.dto.dto.FacilityQuantityDto;
import octguy.livanabe.dto.dto.ImageOrderResponse;
import octguy.livanabe.dto.dto.ListingHostDto;

import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

@Data
@Builder
public class HomeListingResponse {

    private UUID listingId;

    private ListingHostDto host;

    private String title;

    private BigDecimal price;

    private String description;

    private int capacity;

    private String address;

    private Double latitude;

    private Double longitude;

    private UUID propertyTypeId;

    private List<UUID> amenityIds;

    private List<FacilityQuantityDto> facilities;

    private List<ImageOrderResponse> images;
}
